app/src/main/java/com/hedvig/app/feature/crossselling/ui/CrossSellingResultActivity.kt

```diff
@@ -0,0 +1,43 @@
+package com.hedvig.app.feature.crossselling.ui
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.os.Parcelable
+import androidx.activity.compose.setContent
+import com.hedvig.app.BaseActivity
+import kotlinx.parcelize.Parcelize
+import java.time.LocalDate
+
+class CrossSellingResultActivity : BaseActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val crossSellingResult: CrossSellingResult =
+            intent.getParcelableExtra(CROSS_SELLING_RESULT) ?: CrossSellingResult.Error
```

Comment:
Don't think the the case that the thing wouldn't exist inside the parcelable is even possible, please do comment if we want to handle this somewhat differently

Reply:
I think it's fine to throw an error here, as there's a clear 'misuse' of the `Activity` if there is no `CROSS_SELLING_RESULT` on the `Intent`.

Something like this, perhaps:

```kotlin
val crossSellingResult = intent.getParcelableExtra<CrossSellingResult>(CROSS_SELLING_RESULT) ?: throw IllegalArgumentException("Programmer error: CROSS_SELLING_RESULT not provided to ${this.javaClass.name}")
```